tsflex/features/feature_collection.py

```diff
@@ -341,6 +364,63 @@ def _process_segment_idxs(
             segment_idxs = segment_idxs.squeeze()  # remove singleton dimensions
         return segment_idxs
 
+    def _calculate_group_by(
+        self,
+        data: Union[pd.Series, pd.DataFrame, List[Union[pd.Series, pd.DataFrame]]],
+        group_by: str,
+        return_df: Optional[bool] = False,
+        **calculate_kwargs,
+    ):
+        series_list = to_series_list(data)
+        series_names = [s.name for s in series_list]
+        # Check if the group_by column is a valid column
+        assert (
+            group_by in series_names
+        ), f"Data contains no column named '{group_by}' to group by."
+        # now extract all corresponding rows into separate series
+
+        # get all unique values from group_by column
+        group_by_unique_values = list(
+            filter(lambda n: group_by == n.name, series_list)
+        )[0].unique()
+
+        # for convenience purposes, turn series_list into a DF
+        df = pd.concat(series_list, axis=1)
```

Comment:
So an index-aware concatenation is performed, This implies that we will only have group labels for values whose index match *exactly* those of the group-column!

This is really important to document and showcase.